### ISSUE TITLE
Fuse `label_new_pr` and `label_pr` workflows

### ISSUE THREAD

**Issue by dhruvkb** [WordPress/openverse#897]
## Historical context

The task of labelling a new PR is divided into two workflows because of pre-monorepo reasons. Because the Python code to fetch the related issue and detect labels was located in the `WordPress/openverse` repo, it had to invoked remotely from all others repos.

The `label_new_pr.yml` workflow was synced to all repositories, while the `label_pr.yml` workflow was centralised. The latter was called by the former for all new PRs with info about the PR.

## Problem

These workflows are extremely coupled and should be unified into one when there is no multi-repo complexity left. Since the labelling Python code would live in the same repo as all PRs the workflow can directly run it for all incoming PRs.

**Comment by dhruvkb** [comment]
This is <strong>⛔ blocked</strong> till the monorepo project is completed.

**Comment by dhruvkb** [comment]
This issue is still blocked because the monorepo does not include the WordPress/openverse-infrastructure repo which still uses the `label_new_pr` workflow. When it is unblocked, #3125 must be reopened and merged.

**Comment by dhruvkb** [comment]
With #3443, we decided to abandon the two separate workflows and now PR labelling is handled by the "PR automations" workflow.

https://github.com/WordPress/openverse/blob/5f4a991472ffa6eca9f02cd272444932fba9bd25/.github/workflows/pr_automations.yml#L68
